controllers/preflightvalidationocp_reconciler_test.go

```diff
@@ -0,0 +1,231 @@
+				mockRegistry.EXPECT().GetHeaderDataFromLayer(nil, driverToolkitJSONFilePath).Return(dtkDataBytes, nil),
+			)
+
+			res1, res2, err := pro.getKernelVersionAndOSFromDTK(ctx, "dtkImage")
+
+			Expect(err).To(BeNil())
+			Expect(res1).To(Equal("kernelVersion"))
+			Expect(res2).To(Equal("rhelVersion"))
+		})
+
+		It("etc/driver-toolkit-release.json not present in dtk", func() {
+			digests := []string{"digest1", "digest2"}
+			gomock.InOrder(
+				mockRegistry.EXPECT().GetLayersDigests(ctx, "dtkImage", nil, mockAuth).Return(digests, &registry.RepoPullConfig{}, nil),
+				mockRegistry.EXPECT().GetLayerByDigest(digests[1], &registry.RepoPullConfig{}).Return(nil, nil),
+				mockRegistry.EXPECT().GetHeaderDataFromLayer(nil, driverToolkitJSONFilePath).Return(nil, fmt.Errorf("some error")),
+				mockRegistry.EXPECT().GetLayerByDigest(digests[0], &registry.RepoPullConfig{}).Return(nil, nil),
+				mockRegistry.EXPECT().GetHeaderDataFromLayer(nil, driverToolkitJSONFilePath).Return(nil, fmt.Errorf("some error")),
+			)
+
+			_, _, err := pro.getKernelVersionAndOSFromDTK(ctx, "dtkImage")
+
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("etc/driver-toolkit-release.json invalid format", func() {
+			digests := []string{"digest1", "digest2"}
+			dtkReleaseData.KernelVersion = ""
+			dtkDataBytes, err := json.Marshal(&dtkReleaseData)
+			Expect(err).To(BeNil())
+			gomock.InOrder(
+				mockRegistry.EXPECT().GetLayersDigests(ctx, "dtkImage", nil, mockAuth).Return(digests, &registry.RepoPullConfig{}, nil),
+				mockRegistry.EXPECT().GetLayerByDigest(digests[1], &registry.RepoPullConfig{}).Return(nil, nil),
+				mockRegistry.EXPECT().GetHeaderDataFromLayer(nil, driverToolkitJSONFilePath).Return(dtkDataBytes, nil),
+			)
+
+			_, _, err = pro.getKernelVersionAndOSFromDTK(ctx, "dtkImage")
+
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("preparePreflightValidation", func() {
+		var (
+			releaseOCPData openapivi.ImageStream
+			dtkReleaseData dtkRelease
+			pvo            kmmv1beta1.PreflightValidationOCP
+		)
+		BeforeEach(func() {
+			dtkReleaseData = dtkRelease{KernelVersion: "kernelVersion", RTKernelVersion: "rtKernelVersion", RHELVersion: "rhelVersion"}
+			releaseOCPData = openapivi.ImageStream{
+				Spec: openapivi.ImageStreamSpec{
+					Tags: []openapivi.TagReference{
+						{
+							Name: driverToolkitSpecName,
+							From: &corev1.ObjectReference{Name: dtkImageReference},
+						},
+						{
+							Name: "some other component",
+							From: &corev1.ObjectReference{Name: "some other component image"},
+						},
+					},
+				},
+			}
+			pvo = kmmv1beta1.PreflightValidationOCP{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pvo name",
+					Namespace: "pvo namespace",
+				},
+				Spec: kmmv1beta1.PreflightValidationOCPSpec{
+					ReleaseImage:   "ocpReleaseImage",
+					PushBuiltImage: true,
+				},
+			}
+		})
+
+		It("good flow", func() {
```

Comment:
Would we like to have also a couple of the errored flows of this method tested? For example, we could test an error on [this](https://github.com/rh-ecosystem-edge/kernel-module-management/pull/262/files#diff-896d99d61b4773ad699ff552b435c9e1a4622f44edd4bc1a87d1e7151f527596R172-R175) and [this](https://github.com/rh-ecosystem-edge/kernel-module-management/pull/262/files#diff-896d99d61b4773ad699ff552b435c9e1a4622f44edd4bc1a87d1e7151f527596R180-R182). WDYT?

PS. I realize that it's not straight forward, since we are mocking packages that are not directly called by `preparePreflightValidation()`, but I think it would make sense to have the errored flows tested as well.

Reply:
We have an extensive error flow for both getKernelVersionAndOSFromDTK and getDTKFromImage ( since most of the extraction and substition logic is there.  preparePreflightValidation is pretty straightforward, so i think that only verifying good flow should be sufficient ( it is done mainly to test that we are calling getKernelVersionAndOSFromDTK, getDTKFromImage and kernelOsDtkMapping in correct order

Reply:
I understand your point and it is minor and straight forward, but it would make it easier for another developer to contribute to `preparePreflightValidation()` if needed.

> Just as a minor comment, anyhow it is uncommon to test private methods/functions, but I do get why we're doing it here.